backend/x86_binary_emitter.ml

```diff
@@ -420,36 +420,38 @@ let buf_sym b sym offset =
       record_reloc b (Buffer.length b.buf) (Relocation.Kind.DIR32 (lbl, offset));
       buf_int32L b 0L
 
-let emit_mod_rm_reg b rex opcodes rm reg =
+let emit_prefix_modrm b opcodes rm reg ~prefix =
   match rm with
   | Reg32 rm ->
       let rm = rd_of_reg64 rm in
-      emit_rex b (rex lor rexr_reg reg lor rexb_rm rm);
+      prefix b ~rex:0 ~rexr:(rexr_reg reg) ~rexb:(rexb_rm rm) ~rexx:0;
```

Comment:
It is not immediately obvious to me why
"replacing" `rex` with `0` in the `lor` expresion
(not below) is correct. Could you add a
comment?